### PR TITLE
Friendlist and Friendrequests

### PR DESCRIPTION
**Pull request by FurKay00**
- Added endpoints, which allow the frontend to receive the friendlist of a user + received/sent friend requests
- Both sent and received friendrequests come as a whole list and can be seperated through the attribute "status"

### CODE REVIEW

File: src/main/java/de/tinf22b6/dhbwhub/mapper/FriendshipMapper.java
Comment (maxschwinghammer) [inline src/main/java/de/tinf22b6/dhbwhub/mapper/FriendshipMapper.java:1]: Code needs to be formatted correctly, some spaces are missing before and after brackets

File: src/main/java/de/tinf22b6/dhbwhub/service/FriendshipServiceImpl.java [hunk 1]
```diff
@@ -5,11 +5,15 @@
 import de.tinf22b6.dhbwhub.model.Account;
 import de.tinf22b6.dhbwhub.model.Friendship;
 import de.tinf22b6.dhbwhub.proposal.FriendshipProposal;
+import de.tinf22b6.dhbwhub.proposal.simplifiedModels.FriendlistProposal;
+import de.tinf22b6.dhbwhub.proposal.simplifiedModels.FriendrequestProposal;
 import de.tinf22b6.dhbwhub.repository.FriendshipRepository;
 import de.tinf22b6.dhbwhub.service.interfaces.FriendshipService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
```
Comment (maxschwinghammer) [inline src/main/java/de/tinf22b6/dhbwhub/service/FriendshipServiceImpl.java:14]: Can be simplified with `import java.util.*;`

File: src/main/java/de/tinf22b6/dhbwhub/service/interfaces/FriendshipService.java [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import de.tinf22b6.dhbwhub.model.Friendship;
 import de.tinf22b6.dhbwhub.proposal.FriendshipProposal;
+import de.tinf22b6.dhbwhub.proposal.simplifiedModels.FriendlistProposal;
+import de.tinf22b6.dhbwhub.proposal.simplifiedModels.FriendrequestProposal;
```
Comment (maxschwinghammer) [inline src/main/java/de/tinf22b6/dhbwhub/service/interfaces/FriendshipService.java:7]: Simplification possible

**Review by maxschwinghammer — CHANGES_REQUESTED** [review]
@FurKay00 Left a few small comments

**Review by denniskp — APPROVED** [review]

**Review by Nahlam4 — APPROVED** [review]
ich mag die formatierung!

**Review by denniskp — APPROVED** [review]
Die Formatierung ist perfekt

**Pull request merged**